nrf_security/src/zephyr/CMakeLists.txt

```diff
@@ -64,8 +64,19 @@ if(DEFINED src_zephyr)
     $<TARGET_PROPERTY:${mbedcrypto_target},INTERFACE_COMPILE_DEFINITIONS>
   )
 
-  zephyr_library_sources_ifdef(CONFIG_USERSPACE mbedtls_partition.c)
+  # When CONFIG_MBEDLTS is enabled Zephyr defines the MbedTLS
+  # partition in zephyr/kernel/userpace.c
+  # When nrf_security is enabled it provides an implementation
+  # of the the MbedTLS library. Since the MbedTLS library requires
+  # the k_mbedtls_partition when the userspace is enabled
+  # (see /doc/kernel/usermode/memory_domain.rst)
+  # we need to define it here.
+  if(NOT CONFIG_MBEDTLS AND CONFIG_USERSPACE)
```

Comment:
Based on the above comment, shouldn't this be

if ( NOT (CONFIG_MBEDTLS AND CONFIG_USERSPACE))

When nrf_security is enabled we always need something to provide this partition.

When CONFIG_MBEDTLS AND CONFIG_USERSPACE is enabled this is provided by zephyr, otherwise we provide it.

Reply:
I think that the comment in Zephyr is clear that this is used when the userspace is enabled. What you are proposing enables it even if USERSPACE is disabled. So I will not follow this.

Reply:
I added the parenthesis to make it more clear